onnxruntime/core/providers/trt/trt_execution_provider.cc

```diff
@@ -0,0 +1,420 @@
+        // if the output is connected to nodes that don't belong to the subgraph, the output need to be added
+        // to the output list
+        if (node->GetOutputEdgesCount() > node->OutputDefs().size()) {
+          for (auto it = node->OutputEdgesBegin(); it != node->OutputEdgesEnd(); ++it) {
+            const auto& node_index = it->GetNode().Index();
+            const auto& output = (it->GetNode()).InputDefs()[it->GetDstArgIndex()];
+
+            if (node_set.find(node_index) != node_set.end()) {
+              const auto& iter = fused_inputs.find(output);
+
+              if (iter != fused_inputs.end()) {
+                fused_inputs.erase(iter);
+                erased.insert(output);
+              } else if (erased.find(output) == erased.end()) {
+                fused_outputs[output] = output_order++;
+              }
+            } else {
+              fused_outputs_to_add[output] = output_order++;
+            }
+          }
+        } else {
+          for (const auto& output : node->OutputDefs()) {
+            const auto& it = fused_inputs.find(output);
+
+            if (it != fused_inputs.end()) {
+              fused_inputs.erase(it);
+              erased.insert(output);
+            }
+            // only when output is neither in input list nor erased list, add the output to output list
+            else if (erased.find(output) == erased.end()) {
+              fused_outputs[output] = output_order++;
+            }
+          }
+        }
+      }
+
+      fused_outputs.insert(fused_outputs_to_add.begin(), fused_outputs_to_add.end());
+
+      // Sort inputs and outputs by the order they were added
+      std::multimap<int, const NodeArg*> inputs, outputs;
+
+      for (auto it = fused_inputs.begin(); it != fused_inputs.end(); ++it) {
+        inputs.insert(std::pair<int, const NodeArg*>(it->second, it->first));
+      }
+
+      for (auto it = fused_outputs.begin(); it != fused_outputs.end(); ++it) {
+        outputs.insert(std::pair<int, const NodeArg*>(it->second, it->first));
+      }
+
+      // Assign inputs and outputs to subgraph's meta_def
+      auto meta_def = std::make_unique<::onnxruntime::IndexedSubGraph::MetaDef>();
+      meta_def->name = "TRTKernel";
+      meta_def->domain = kMSDomain;
+
+      for (const auto& input : inputs) {
+        meta_def->inputs.push_back(input.second->Name());
+      }
+
+      for (const auto& output : outputs) {
+        meta_def->outputs.push_back(output.second->Name());
+      }
+
+      meta_def->since_version = 1;
+      sub_graph->SetMetaDef(meta_def);
+    }
+
+    result.push_back(std::make_unique<ComputeCapability>(std::move(sub_graph)));
+  }
+
+  return result;
+}
+
+std::shared_ptr<KernelRegistry> TRTExecutionProvider::GetKernelRegistry() const {
+  return nullptr;
+}
+
+common::Status TRTExecutionProvider::CopyTensor(const Tensor& src, Tensor& dst) const {
```

Comment:
>CopyTensor [](start = 37, length = 10)

Does the CUDAExecutionProvider get used to to the copy to/from device so we don't need equiavalent logic here?

Reply:
Right now for TRT kernel the default memory type is CPU. CUDA<->CPU memory copy is done inside of TRT compute function by calling CUDA APIs.